### PR TITLE
PMM-6968 Notification channels API

### PR DESCRIPTION
**Pull request by artemgavrilov**
https://jira.percona.com/browse/PMM-6968

- [ ] https://github.com/percona/pmm-managed/pull/532

### CODE REVIEW

File: management/ia/notifications_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		require.NoError(t, err)
+
+		_, err = client.RemoveChannel(&channels.RemoveChannelParams{
+			Body: channels.RemoveChannelBody{
+				ChannelID: id,
+			},
+			Context: pmmapitests.Context,
+		})
+
+		resp, err := client.ListChannels(&channels.ListChannelsParams{Context: pmmapitests.Context})
```
Comment (palash25) [inline management/ia/notifications_test.go:137]: nit i know we are already testing it here, but I feel like we should have a separate test for `ListChannels`

File: management/ia/notifications_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+
+		_, err = client.RemoveChannel(&channels.RemoveChannelParams{
+			Body: channels.RemoveChannelBody{
+				ChannelID: id,
+			},
+			Context: pmmapitests.Context,
+		})
```
Comment (palash25) [inline management/ia/notifications_test.go:135]: shouldn't there be a `require.NoError(t, err)` after this ?

File: management/ia/notifications_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+				assert.True(t, channel.EmailConfig.SendResolved)
+				found = true
+			}
+		}
+
+		assert.True(t, found, "Expected channel not found")
+	})
+}
+
+func TestRemoveChannel(t *testing.T) {
```
Comment (palash25) [inline management/ia/notifications_test.go:112]: we should also add one test case where `RemoveChannel`  errors

File: management/ia/notifications_test.go (hunk 1 above)
Comment (artemgavrilov) [reply management/ia/notifications_test.go:150]: Added

File: management/ia/notifications_test.go (hunk 2 above)
Comment (artemgavrilov) [reply management/ia/notifications_test.go:147]: Yep, fixed

File: management/ia/notifications_test.go (hunk 3 above)
Comment (artemgavrilov) [reply management/ia/notifications_test.go:120]: Done

**Review by BupycHuk — APPROVED** [review]

**Review by palash25 — APPROVED** [review]

File: management/ia/notifications_test.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+func TestAddChannel(t *testing.T) {
+	if !pmmapitests.RunIATests {
+		t.Skip("Skipping IA tests until IA will out of beta: https://jira.percona.com/browse/PMM-7001")
+	}
+
+	client := channelsClient.Default.Channels
+
+	t.Run("normal", func(t *testing.T) {
+		_, err := client.AddChannel(&channels.AddChannelParams{
```
Comment (palash25) [inline management/ia/notifications_test.go:24]: since we are returning ID now we should check this value whether it is empty or not

Comment (artemgavrilov) [reply management/ia/notifications_test.go]: Done

**Review by palash25 — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**